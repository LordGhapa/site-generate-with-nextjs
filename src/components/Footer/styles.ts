/* eslint-disable no-unused-vars */
import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
${({ theme }) => css`

text-align: center;
font-size: ${theme.font.sizes.small};
border-top: .2rem solid ${theme.colors.mediumGray};
padding-block: 3rem;
background-color: #fff;
a{
  color: inherit;
  text-decoration: none;
}
& ${TextComponent}{
  font-size:${theme.font.sizes.small};

}
${SectionContainer}{
  padding: 0;
}

`}
`;
