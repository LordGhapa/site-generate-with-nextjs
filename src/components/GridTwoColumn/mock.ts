import { GridTwoColumnProps } from '.';

export default {
  title: 'Grid Two Columns',
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Facere, quia odio! Harum, doloribus. Libero, ratione! Natus, sequi iure delectus, a ducimus dolorem obcaecati dolore, ullam quibusdam dolores et odio exercitationem.',
  srcImg: 'assets/images/javascript.svg',
} as GridTwoColumnProps;
