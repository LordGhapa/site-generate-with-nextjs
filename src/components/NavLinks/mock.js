export default [
  {
    children: 'link 1',
    link: 'target 1',
    newTab: false,
  },
  {
    children: 'link 2',
    link: 'target 2',
    newTab: false,
  },
  {
    children: 'link 3',
    link: 'target 3',
    newTab: false,
  },
];
