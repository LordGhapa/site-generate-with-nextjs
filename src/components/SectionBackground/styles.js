/* eslint-disable no-unused-vars */

import styled, { css } from 'styled-components';

const containerBackgroundActivate = (theme) => css`
color: ${theme.colors.white};
background: ${theme.colors.primaryColor};
`;

export const Container = styled.div`
${({ theme, background }) => css`
background: ${theme.colors.white};
color: ${theme.colors.primaryColor};


${background && containerBackgroundActivate(theme)};
min-height: 100vh;
display: flex;
align-items: center;
justify-content: center;
`}
`;
