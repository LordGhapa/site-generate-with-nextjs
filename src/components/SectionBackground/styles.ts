/* eslint-disable no-unused-vars */

import styled, { DefaultTheme, css } from 'styled-components';
type bg = {
  background: boolean;
};

const containerBackgroundActivate = (theme: DefaultTheme) => css`
color: ${theme.colors.white};
background: ${theme.colors.primaryColor};
`;

export const Container = styled.div<bg>`
${({ theme, background }) => css`
background: ${theme.colors.white};
color: ${theme.colors.primaryColor};


${background && containerBackgroundActivate(theme)};
min-height: 100vh;
display: flex;
align-items: center;
justify-content: center;
`}
`;
