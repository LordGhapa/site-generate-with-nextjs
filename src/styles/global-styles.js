import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
*{
 margin:0;
 padding: 0;
 box-sizing: border-box;
}
html{
  font-size: 62.5%;
  scroll-behavior: smooth;
}
body{
  font-family: 'Open Sans', sans-serif;
  background-color: gray;
  font-family: ${({ theme }) => theme.font.family.default};
}
h1,h2,h3,h4,h5,h6{
  margin: ${({ theme }) => theme.spacings.sizes.large} 0;
}
p{
  margin: ${({ theme }) => theme.spacings.sizes.medium} 0;
}
ul,ol{
  margin: ${({ theme }) => theme.spacings.sizes.medium};
  padding: ${({ theme }) => theme.spacings.sizes.medium};
}
a{
  color: ${({ theme }) => theme.colors.secondaryColor};
}
`;
