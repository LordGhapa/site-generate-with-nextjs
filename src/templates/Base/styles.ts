/* eslint-disable no-unused-vars */
import styled, { css } from 'styled-components';

export const Container = styled.div`
${({ theme }) => css`

@media ${theme.media.lteMedium} {
  padding-top: 0;
}
`}
`;
