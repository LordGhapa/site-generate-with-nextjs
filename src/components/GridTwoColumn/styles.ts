/* eslint-disable no-unused-vars */
import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
${({ theme }) => css`
display: grid;
grid-template-columns: 1fr 1.5fr;
align-items: center;
gap: ${theme.spacings.sizes.large};
 img{
  width: 100%;

}
@media ${theme.media.lteMedium}{
  grid-template-columns: 1fr ;
}

${Title}{
  margin-bottom: ${theme.spacings.sizes.large};

}
`}
`;
export const TextContainer = styled.div`
${({ theme }) => css`

@media ${theme.media.lteMedium}{
 margin-bottom:${theme.spacings.sizes.large};
}

`}
`;

export const ImageContainer = styled.div`
${() => css``}
`;
export const Image = styled.img`
${() => css``}
`;
