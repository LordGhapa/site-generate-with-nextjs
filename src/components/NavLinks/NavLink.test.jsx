import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { NavLinks } from '.';
import mock from './mock';

describe('<NavLink />', () => {
  it('should render NavLink with mock', () => {
    renderTheme(<NavLinks links={mock} />);
    expect(screen.getAllByRole('link')).toHaveLength(mock.length);
  });

  it('should NOT render NavLink', () => {
    renderTheme(<NavLinks />);
    expect(screen.queryAllByText(/link/i)).toHaveLength(0);
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(<NavLinks links={mock} />);

    expect(container).toMatchSnapshot();
  });
});
