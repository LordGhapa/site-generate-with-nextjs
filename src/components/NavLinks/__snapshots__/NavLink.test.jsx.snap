// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c1 {
  font-size: 1.6rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c1::after {
  content: "";
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0%;
  height: 0.2rem;
  background-color: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c1:hover::after {
  left: 25%;
  width: 50%;
}

@media ((max-width:768px)) {
  .c0 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

<div>
  <nav
    aria-label="Main Menu"
    class="c0"
  >
    <a
      class="c1"
      href="target 1"
      target="_self"
    >
      link 1
    </a>
    <a
      class="c1"
      href="target 2"
      target="_self"
    >
      link 2
    </a>
    <a
      class="c1"
      href="target 3"
      target="_self"
    >
      link 3
    </a>
  </nav>
</div>
`;
